util/util.go

```diff
@@ -388,16 +404,52 @@ func referSameObject(a, b metav1.OwnerReference) bool {
 }
 
 // PointsTo returns true if any of the owner references point to the given target
+// Deprecated: Use PointsToObject to cover differences in API version or backup/restore that changed UIDs
 func PointsTo(refs []metav1.OwnerReference, target *metav1.ObjectMeta) bool {
 	for _, ref := range refs {
 		if ref.UID == target.UID {
 			return true
 		}
 	}
+	return false
+}
 
+// PointsToObject returns true if any of the owner references point to the given target
+func PointsToObject(obj metav1.Object, target Object) bool {
+	for _, ref := range obj.GetOwnerReferences() {
+		ref := ref
+		if refersTo(&ref, target) {
+			return true
+		}
+	}
 	return false
 }
 
+type Object interface {
+	metav1.Object
+	GroupVersionKind() schema.GroupVersionKind
+}
+
+// IsControlledBy differs from metav1.IsControlledBy in that it checks the group (but not version), kind, and name vs uid
+func IsControlledBy(obj metav1.Object, owner Object) bool {
+	controllerRef := metav1.GetControllerOfNoCopy(obj)
+	if controllerRef == nil {
+		return false
+	}
+	return refersTo(controllerRef, owner)
+}
+
+// Returns true if ref refers to obj
+func refersTo(ref *metav1.OwnerReference, obj Object) bool {
```

Comment:
This internal method and the other one are almost the same :/

Reply:
Yeah, interface vs struct in the second position. I'll move 'em next to each other at least.